java/ql/src/Security/CWE/CWE-502/UnsafeDeserialization.ql

```diff
@@ -54,6 +62,27 @@ class UnsafeDeserializationConfig extends TaintTracking::Configuration {
       ma.getArgument(0) = node.asExpr() and
       exists(SafeJsonIoConfig sji | sji.hasFlowToExpr(ma.getArgument(1)))
     )
+    or
+    exists(MethodAccess ma |
+      (
+        ma.getMethod() instanceof FlexjsonDeserializeMethod or
+        ma.getMethod() instanceof JoddJsonParseMethod
+      ) and
+      node.asExpr() = ma.getAnArgument() and
+      (
+        ma.getArgument(1).(TypeLiteral).getType() instanceof NullType // Require that the class parameter is non-null
```

Comment:
The comment says "non-null", but this checks for `null`.
Also it might be easier to check for `NullLiteral` instead; not sure if `(TypeLiteral).getType() instanceof NullType` is even possible because a type literal cannot represent `null`.

Reply:
Changed